lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -274,7 +308,9 @@ sub _get_run_last_info {
 
     my $linkpath = Mojo::File->new($home, $project, $batch, '.run_last');
     my $folder;
-    eval { $folder = readlink($linkpath) };
+    eval { $folder = readlink($linkpath); 1; }
+      or log_error("Cannot read symbolic link in _get_run_last_info: $_");
```

Comment:
Probably should use `$@` too.

Reply:
Actually, `readlink` usually does not raise errors but return an undef value (with `$!`). So this code is wrong.

Reply:
Exactly - usually. And if it 'sometimes' throws an exception - it is enough reason to wrap it with eval, otherwise UI will not load at all in those cases instead of showing what it can.